src/components/v2/chartDeploymentHistory/ChartDeploymentHistory.component.tsx

```diff
@@ -0,0 +1,577 @@
+import { ServerErrors } from '../../../modals/commonTypes'
+import { Moment12HourFormat, URLS } from '../../../config'
+import CodeEditor from '../../CodeEditor/CodeEditor'
+import moment from 'moment'
+import Tippy from '@tippyjs/react'
+import '../../app/details/cIDetails/ciDetails.scss'
+import YAML from 'yaml'
+import './chartDeploymentHistory.scss'
+import MessageUI from '../common/message.ui'
+import { toast } from 'react-toastify'
+import { useHistory, useRouteMatch } from 'react-router'
+import CDEmptyState from '../../app/details/cdDetails/CDEmptyState'
+import {
+    ChartDeploymentDetail,
+    ChartDeploymentHistoryResponse,
+    ChartDeploymentManifestDetail,
+    getDeploymentHistory,
+    getDeploymentManifestDetails,
+    rollbackApplicationDeployment,
+    RollbackReleaseRequest,
+} from './chartDeploymentHistory.service'
+import IndexStore from '../appDetails/index.store'
+
+interface DeploymentManifestDetail extends ChartDeploymentManifestDetail {
+    loading?: boolean
+    error?: boolean
+    errorCode?: number
+    isApiCallInProgress?: boolean
+}
+
+function ChartDeploymentHistory({
+    appId,
+    appName,
+    isExternal,
+}: {
+    appId: string
+    appName?: string
+    isExternal: boolean
+}) {
+    const [isLoading, setIsLoading] = useState(true)
+    const [errorResponseCode, setErrorResponseCode] = useState(undefined)
+    const [deploymentHistoryArr, setDeploymentHistoryArr] = useState<ChartDeploymentDetail[]>([])
+    const [installedAppInfo, setInstalledAppInfo] = useState<InstalledAppInfo>()
+    const [selectedDeploymentHistoryIndex, setSelectedDeploymentHistoryIndex] = useState<number>(0)
+    const [selectedDeploymentTabIndex, setSelectedDeploymentTabIndex] = useState<number>(0)
+    const [deploymentManifestDetails, setDeploymentManifestDetails] = useState<Map<number, DeploymentManifestDetail>>()
+    const [rollbackDialogTitle, setRollbackDialogTitle] = useState('Rollback')
+    const [showRollbackConfirmation, setShowRollbackConfirmation] = useState(false)
+    const [deploying, setDeploying] = useState(false)
+    const history = useHistory()
+    const { url } = useRouteMatch()
+
+    const deploymentTabs: string[] = isExternal
+        ? ['Source', 'values.yaml', 'Helm generated manifest']
+        : ['Source', 'values.yaml']
+
+    // component load
+    useEffect(() => {
+        console.log(IndexStore.getAppDetails())
+        getDeploymentHistory(appId, isExternal)
+            .then((deploymentHistoryResponse: ChartDeploymentHistoryResponse) => {
+                const _deploymentHistoryArr =
+                    deploymentHistoryResponse.result?.deploymentHistory?.sort(
+                        (a, b) => b.deployedAt.seconds - a.deployedAt.seconds,
+                    ) || []
+                setDeploymentHistoryArr(_deploymentHistoryArr)
+                setInstalledAppInfo(deploymentHistoryResponse.result?.installedAppInfo)
```

Comment:
Same here

Reply:
result in the response has been marked as optional at a few places. This needs to be checked why and fix it for all at once.

Reply:
As discussed, this is something that needs to be confirmed and changed at different places. So, keeping it as is for now.